Lib/random.py

```diff
@@ -379,8 +383,20 @@ def sample(self, population, k):
             population = tuple(population)
         if not isinstance(population, _Sequence):
             raise TypeError("Population must be a sequence.  For dicts or sets, use sorted(d).")
-        randbelow = self._randbelow
         n = len(population)
+        if weights is not None:
+            cum_weights = list(_accumulate(weights))
+            if len(cum_weights) != n:
+                raise ValueError('The number of weights does not match the population')
+            total = cum_weights.pop()
+            if not isinstance(total, int):
+                raise TypeError('Weights must be integers')
+            if total < 0:
+                raise ValueError('Total of weights must be greater than zero')
```

Comment:
Should this be `if total <= 0`?

If the weight are all zero, for example:

```python
random.sample(population=["red"], k=1, weights=[0])
```

you get a misleading error message:

```
ValueError: Sample larger than population or is negative
```

Reply:
Okay.